src/IQFeed.CSharpApiClient/Streaming/Level1/Level1Snapshot.cs

```diff
@@ -59,12 +61,20 @@ void Level1ClientOnFundamental(FundamentalMessage fundamentalMessage)
                     res.TrySetResult(fundamentalMessage);
             }
 
+            void Level1ClientOnSymbolNotFound(SymbolNotFoundMessage symbolNotFoundMessage)
+            {
+                if (symbolNotFoundMessage.Symbol == symbol)
+                    res.TrySetException(new SymbolNotFoundIQFeedException(symbol));
+            }
+
             _level1MessageHandler.Fundamental += Level1ClientOnFundamental;
+            _level1MessageHandler.SymbolNotFound += Level1ClientOnSymbolNotFound;
             ReqWatch(symbol);
```

Comment:
what do you think if we return the request as a string from ReqWatch like
```
private string ReqWatch(string symbol)
        {
            var request = _level1RequestFormatter.ReqWatch(symbol);
            _socketClient.Send(request);
            return request;
        }
```

that way, we can reuse the request in the exception to make sure we have something consistent for all exceptions

same for ReqUnwatch

Reply:
@mrut2pac could you please also add a quick integration test. this is quickly achieve.

Reply:
That's what I was doing in the beginning, but then I realized the delegate is set before calling ReqWatch, therefore it need to be either handled in ReqWatch or we need to break it down. Let me see what's the best way of doing it.
We are not handling any errors for ReqUnwatch, do we need it?
Adding a test is a good idea. I will do it.

Reply:
ahh yes you're right. Than, maybe just call the requestFormatter in the delegate...

Reply:
split the request creation and sending to get more elegant solution

Reply:
perfect!